File: DSRFQ.Web/Modules/Company/CompanyGroups/CompanyGroupsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace DSRFQ.Company.Columns;

[ColumnsScript("Company.CompanyGroups")]
[BasedOnRow(typeof(CompanyGroupsRow), CheckNames = true)]
public class CompanyGroupsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string CompanyName { get; set; }
    public string GroupName { get; set; }
    
}

File: DSRFQ.Web/Modules/ServerTypes/Company/CompanyGroupsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CompanyGroupsRow } from "./CompanyGroupsRow";

export interface CompanyGroupsColumns {
    Id: Column<CompanyGroupsRow>;
    CompanyName: Column<CompanyGroupsRow>;
    GroupName: Column<CompanyGroupsRow>;
}

export class CompanyGroupsColumns extends ColumnsBase<CompanyGroupsRow> {
    static readonly columnsKey = 'Company.CompanyGroups';
    static readonly Fields = fieldsProxy<CompanyGroupsColumns>();
}